### ISSUE TITLE
Increase the idleTimeout to 15 seconds to enhance API response time.

### ISSUE THREAD

**Issue by nahidmk** [cfug/dio#2046]
### Request Statement

I want to improve the response time of my API call. When I hit an API using Postman, it takes only 80ms, but the same API takes 250ms in Flutter Dio. Some people suggested me to increase idleTimeout to 15 seconds to improve API response time. I have changed idleTimeout from 3 seconds to 15 seconds, and it actually improves the API response time. So, does it cause any problems to increase idleTimeout to 15 seconds?

**Comment by AlexV525** [comment]
Which `idleTimeout`?

**Comment by nahidmk** [comment]
inside the io_adapter

HttpClient _createHttpClient() {
    if (createHttpClient != null) {
      return createHttpClient!();
    }
    final client = HttpClient()..idleTimeout = const Duration(seconds: 15);
    // ignore: deprecated_member_use, deprecated_member_use_from_same_package
    return onHttpClientCreate?.call(client) ?? client;
  }

**Comment by AlexV525** [comment]
In the code you referenced, you can always specify the `idleTimeout` when using `createHttpClient`.

**Comment by AlexV525** [comment]
Improvements are related to use cases. Your use case might fit the 15 seconds idle timeout perfectly, others might not. Generally, changing such configuration will break users code, no one want that happen. Also, as I stated above https://github.com/cfug/dio/issues/2046#issuecomment-1827144807, there is no need for you to change the source code. Instead, set the `createHttpClient` to create your 15 secs timeout client.

Closing as we've already provide such ability.
